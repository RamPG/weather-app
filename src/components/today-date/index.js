import TodayDate from './today-date';

export default TodayDate;
