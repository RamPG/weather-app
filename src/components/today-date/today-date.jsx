import React from 'react';

import './today-date.scss';
import {
  getWeekDayNow, getMonthDayNow, getYearNow, getMonthNow, getNameDay, getNameMonth,
} from '../../services/time-library';

const TodayDate = () => (
  <h1>
    {`Today is ${getMonthDayNow()} ${getNameMonth(getMonthNow())} ${getYearNow()} ${getNameDay(getWeekDayNow())}`}
  </h1>

);
export default TodayDate;
