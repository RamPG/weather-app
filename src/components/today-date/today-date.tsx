import React, { FunctionComponent } from 'react';

import './today-date.scss';

export type TodayDatePropsType = {
  monthDay: number,
  nameMonth: string,
  year: number,
  nameDay: string
}
export const TodayDate: FunctionComponent<TodayDatePropsType> = (
  {
    monthDay, nameDay, year, nameMonth,
  },
) => (
  <p>
    Today is
    {' '}
    <span>{monthDay}</span>
    {' '}
    <span>{nameMonth}</span>
    {' '}
    <span>{year}</span>
    {' '}
    <span>{nameDay}</span>
  </p>
);
