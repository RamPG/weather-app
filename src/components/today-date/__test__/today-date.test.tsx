import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { TodayDate, TodayDatePropsType } from '../today-date';

describe('Today date', () => {
  let props: TodayDatePropsType;
  beforeAll(() => {
    props = {
      monthDay: 17,
      nameMonth: 'August',
      year: 2020,
      nameDay: 'Monday',
    };
  });
  afterEach(cleanup);
  it('Snapshot', () => {
    const { asFragment } = render(<TodayDate {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
