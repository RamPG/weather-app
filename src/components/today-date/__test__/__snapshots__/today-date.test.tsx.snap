// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Today date Snapshot 1`] = `
<DocumentFragment>
  <p>
    Today is 
    <span>
      17
    </span>
     
    <span>
      August
    </span>
     
    <span>
      2020
    </span>
     
    <span>
      Monday
    </span>
  </p>
</DocumentFragment>
`;
